### PR TITLE
Distro

### PR DESCRIPTION
**Pull request by alexeagle**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,10 +25,11 @@ Next, create a `WORKSPACE` file in your project root (or edit the existing one)
...
 
 ```python
-git_repository(
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
     name = "build_bazel_rules_nodejs",
-    remote = "https://github.com/bazelbuild/rules_nodejs.git",
-    tag = "0.16.8",
+    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/0.16.8/rules_nodejs-0.16.8.tar.gz"],
+    sha256 = "123",
```
Comment (alexeagle) [inline README.md]: don't worry, as soon as this PR lands, I'll cut a release which will update the hash.
We don't have a tar.gz for 0.16.8 anyway.

File: .circleci/config.yml [hunk 2]
```diff
@@ -140,7 +140,21 @@ jobs:
 
       # Not using "bazel run @nodejs//:yarn test" as will attempt to run
       # the "test" script on all folder listed in node_repositories()
-      - run: yarn test
+      - run: yarn test:e2e
```
Comment (gregmagolan) [inline .circleci/config.yml]: nit: should we rename this job to test_e2e?

File: BUILD.bazel [hunk 3]
```diff
@@ -9,3 +12,50 @@ filegroup(
...
+)
+
+npm_package(
+    name = "rules_nodejs_package",
+    srcs = [
+        "BUILD.bazel",
+        "LICENSE",
+        "defs.bzl",
+        "package.bzl",
+        "//third_party/github.com/gjtorikian/isBinaryFile:LICENSE",
```
Comment (gregmagolan) [inline BUILD.bazel]: Wouldn't the LICENSE file be included in the `//third_party/github.com/gjtorikian/isBinaryFile` dep?

File: README.md (hunk 1 above)
Comment (gregmagolan) [reply README.md]: 👍 I also like what rules_webtesting does. In their release notes they include the lines you need to add to your WORKSPACE file to use the version: https://github.com/bazelbuild/rules_webtesting/releases/tag/0.3.0

File: package.json [hunk 4]
```diff
@@ -9,34 +9,32 @@
     "homepage": "https://github.com/bazelbuild/rules_nodejs",
     "license": "Apache-2.0",
     "devDependencies": {
-        "@bazel/bazel": "^0.21.0",
+        "//": "need release candidate of Bazel to pick up https://github.com/bazelbuild/bazel/pull/6980",
+        "@bazel/bazel": "^0.23.0-rc1b",
```
Comment (gregmagolan) [inline package.json]: Nice. That should fix with the rules_typescript pkg_tar issue I was having as well hopefully.

File: internal/web_package/assembler.js [hunk 5]
```diff
@@ -64,7 +64,7 @@ function main(params) {
         copy(path.join(f, file));
       }
     } else {
-      const content = fs.readFileSync(f, {encoding: 'utf-8'});
+      const content = fs.readFileSync(f);
       write(path.join(outdir, relative(f)), content);
```
Comment (gregmagolan) [inline internal/web_package/assembler.js]: Cleaner to just use fs.copyFileSync(src, dest); (same a package.js)

File: on-version.js [hunk 6]
```diff
@@ -1,7 +1,18 @@
...
+// TODO(alexeagle): consider streaming the bytes into the hash function, if this consumes too much
+// RAM
+const sha256 = hash.update(fs.readFileSync(artifact)).digest('hex');
+
 shell.sed('-i', 'VERSION \= \"[0-9\.]*\"', `VERSION = "${version}"`, 'package.bzl');
-shell.sed('-i', 'tag = \"[0-9\.]*\"', `tag = "${version}"`, 'README.md');
+shell.sed(
+    '-i', 'download/[0-9\.]*/rules_nodejs-[0-9\.]*.tar.gz',
+    `download/${version}/rules_nodejs-${version}.tar.gz`, 'README.md');
+shell.sed('-i', 'sha256 = \"[0-9a-f]+\"', `sha256 = "${sha256}"`, 'README.md');
```
Comment (gregmagolan) [inline on-version.js]: 😎

File: BUILD.bazel (hunk 3 above)
Comment (alexeagle) [reply BUILD.bazel]: no, that dep is just a js_library or ts_library

but I'll add a package_contents there so we don't have two references into that directory

File: internal/web_package/assembler.js (hunk 5 above)
Comment (gregmagolan) [reply internal/web_package/assembler.js]: Oh.. but you need to also make sure the folder is there with `mkdirp(path.dirname(dest));`

Comment (alexeagle) [reply internal/web_package/assembler.js]: :+1:

File: package.json [hunk 7]
```diff
@@ -9,34 +9,32 @@
...
         "@bazel/buildifier": "^0.20.0",
-        "@bazel/typescript": "^0.22.1",
+        "@bazel/typescript": "0.23.2",
```
Comment (kyliau) [inline package.json]: 0.23.3

Comment (alexeagle) [reply package.json]: The difference is just the ngtsc hook in ts_library, no need to update to latest

File: README.md (hunk 1 above)
Comment (alexeagle) [reply README.md]: yeah I'd like to automate the entire release, changelog, upload done by the script. next step?

**Pull request merged**